examples/typescript-example/components/with-auth.tsx

```diff
@@ -1,48 +1,49 @@
-import React, { Component } from 'react';
+import React from 'react';
+import { NextPage, NextPageContext } from 'next';
+import { UserProfile, useUser } from '@auth0/nextjs-auth0';
 
 import auth0 from '../lib/auth0';
-import { fetchUser } from '../lib/user';
 import createLoginUrl from '../lib/url-helper';
 import RedirectToLogin from '../components/login-redirect';
 
 type AuthenticatedProps = {
-  user?: any;
-  loading: boolean;
-  children: React.ReactChild | React.ReactChildren;
+  user?: UserProfile;
+  children: React.ReactNode;
```

Comment:
children should optional https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4486794741ccb9f18f562fefade043ac31e817de/types/react/index.d.ts#L822

You could do `type AuthenticatedProps = React.PropsWithChildren & { user?: UserProfile }`

Which should resolve your issues on L32 and L45

Reply:
Changed in https://github.com/auth0/nextjs-auth0/pull/187/commits/5592ca5fe476d1148b8493c7feb81289e251fd15.